rocketpy/simulation/flight.py

```diff
@@ -1422,6 +1429,30 @@ def u_dot(self, t, u, post_processing=False):
             drag_coeff = self.rocket.power_off_drag.get_value_opt(free_stream_mach)
         rho = self.env.density.get_value_opt(z)
         R3 = -0.5 * rho * (free_stream_speed**2) * self.rocket.area * (drag_coeff)
+        for airbrakes in self.rocket.airbrakes:
+            if airbrakes.deployed_level > 0:
+                R3 += (
+                    -0.5
+                    * rho
+                    * (free_stream_speed**2)
+                    * airbrakes.reference_area
+                    * airbrakes.cd(airbrakes.deployed_level, free_stream_mach)
+                )
+                airbrakes.state_history.append(
+                    [
+                        t,
+                        airbrakes.deployed_level,
+                        airbrakes.cd(airbrakes.deployed_level, free_stream_mach),
+                    ]
+                )
+            else:
+                airbrakes.state_history.append(
+                    [
+                        t,
+                        airbrakes.deployed_level,
+                        0,
+                    ]
+                )
```

Comment:
This is a bit of a work around the interpolation/extrapolation of a 2D `Function` object.

Physically speaking, when the deployed_level is zero, there must be no drag from the airbrakes. However, when passing the cd_curve the `Function` class will usually interpolate the data in a way that there will be drag even when deployed level is 0. 

The `Function` class is interpolating correctly, we just have to add this extra check to guarantee physical coherence

Reply:
> Physically speaking, when the deployed_level is zero, there must be no drag from the airbrakes. However, when passing the cd_curve the Function class will usually interpolate the data in a way that there will be drag even when deployed level is 0.

What if the air brakes are set to not have null drag in the default configuration ("openness level"=0%) ?

Is that possible?

Reply:
It is possible, but with our current implementation we assume that a 0% deployed level is the same as the air brakes having no effect in the rocket

I think this is a really reasonable hypothesis, but I will make this clearer in the documentation

Reply:
solved in https://github.com/RocketPy-Team/RocketPy/pull/426/commits/42093674c7432469c0b30c9f786c2c44a1974a5b